### PR TITLE
Spring db configure

### PR DESCRIPTION
**Pull request by jah701**

### CODE REVIEW

**Review by Alkey — APPROVED** [review]

File: src/main/java/com/dev/cinema/model/User.java [hunk 1]
```diff
@@ -20,4 +22,7 @@
     private String password;
     @Column(unique = true)
     private String email;
+    @ManyToMany
+    @Column(name = "roles")
+    private List<Role> roles;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/User.java]: better change to set

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsService implements UserDetailsService {
+    private final UserDao userDao;
+    private final PasswordEncoder encoder;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: do we need this field here?

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Autowired
+    public CustomUserDetailsService(UserDao userDao, PasswordEncoder encoder) {
+        this.userDao = userDao;
+        this.encoder = encoder;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userDao.findByEmail(username).orElseThrow();
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: can we throw UsernameNotFoundException in the orElseThrow()?

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java (hunk 2 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: nit :)

File: src/main/java/com/dev/cinema/model/User.java (hunk 1 above)
Comment (jah701) [reply src/main/java/com/dev/cinema/model/User.java]: Fixed :)

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        this.userDao = userDao;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
+        User user = userDao.findByEmail(username)
+                .orElseThrow(() -> new UsernameNotFoundException("Can't get user wth email: "
+                        + username));
+        UserBuilder builder = null;
+        if (user != null) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: do we need this checking?

Comment (jah701) [reply src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]: Блин, привык на готовеньком сидеть :D

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 5]
```diff
@@ -23,19 +23,14 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
...
-                    .stream()
-                    .map(r -> r.getRole().toString())
-                    .toArray(String[]::new);
-            builder.roles(roles);
-        } else {
-            throw new UsernameNotFoundException("User with email: '"
-                    + username + "' was not found.");
-        }
+        UserBuilder builder;
+        builder = org.springframework.security.core.userdetails.User.withUsername(username);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]:
```suggestion
        UserBuilder builder = org.springframework.security.core.userdetails.User.withUsername(username);
```

File: src/main/java/com/dev/cinema/security/CustomUserDetailsService.java [hunk 6]
```diff
@@ -23,19 +23,14 @@ public UserDetails loadUserByUsername(String username) throws UsernameNotFoundEx
...
+        builder.password(user.getPassword());
+        String[] roles = user.getRoles()
+                .stream()
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/security/CustomUserDetailsService.java]:
```suggestion
        String[] roles = user.getRoles().stream()
```

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**